Nightly performance_test job failing

See for example:

- http://build.ros2.org/view/Fci/job/Fci__nightly-performance_ubuntu_focal_amd64/47/consoleFull#console-section-26

<details>
<summary>Error output</summary>

```
07:20:43 /tmp/ws2/src/performance_test/performance_test/src/communication_abstractions/resource_manager.cpp: In member function ‘eprosima::fastrtps::Participant* performance_test::ResourceManager::fastrtps_participant() const’:
07:20:43 /tmp/ws2/src/performance_test/performance_test/src/communication_abstractions/resource_manager.cpp:81:23: error: ‘class eprosima::fastrtps::rtps::BuiltinAttributes’ has no member named ‘use_SIMPLE_RTPSParticipantDiscoveryProtocol’
07:20:43    81 |   PParam.rtps.builtin.use_SIMPLE_RTPSParticipantDiscoveryProtocol = true;
07:20:43       |                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
07:20:43 /tmp/ws2/src/performance_test/performance_test/src/communication_abstractions/resource_manager.cpp:82:23: error: ‘class eprosima::fastrtps::rtps::BuiltinAttributes’ has no member named ‘use_SIMPLE_EndpointDiscoveryProtocol’
07:20:43    82 |   PParam.rtps.builtin.use_SIMPLE_EndpointDiscoveryProtocol = true;
07:20:43       |                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
07:20:43 /tmp/ws2/src/performance_test/performance_test/src/communication_abstractions/resource_manager.cpp:83:23: error: ‘class eprosima::fastrtps::rtps::BuiltinAttributes’ has no member named ‘m_simpleEDP’
07:20:43    83 |   PParam.rtps.builtin.m_simpleEDP.use_PublicationReaderANDSubscriptionWriter = true;
07:20:43       |                       ^~~~~~~~~~~
07:20:43 /tmp/ws2/src/performance_test/performance_test/src/communication_abstractions/resource_manager.cpp:84:23: error: ‘class eprosima::fastrtps::rtps::BuiltinAttributes’ has no member named ‘m_simpleEDP’
07:20:43    84 |   PParam.rtps.builtin.m_simpleEDP.use_PublicationWriterANDSubscriptionReader = true;
07:20:43       |                       ^~~~~~~~~~~
07:20:43 /tmp/ws2/src/performance_test/performance_test/src/communication_abstractions/resource_manager.cpp:85:23: error: ‘class eprosima::fastrtps::rtps::BuiltinAttributes’ has no member named ‘leaseDuration’
07:20:43    85 |   PParam.rtps.builtin.leaseDuration = eprosima::fastrtps::c_TimeInfinite;
```
</details>

@MiguelCompany the build error seems to come from an API change in `Fast-RTPS`.
Was it intentional and this repo should be updated or should it be fixed in `fastrtps`?

--- a/performance_test/CMakeLists.txt
+++ b/performance_test/CMakeLists.txt
@@ -67,7 +67,7 @@ if(${osrf_testing_tools_cpp_FOUND})
 endif()
 
 # Dashing version looks like 0.7.X.
-if(${rclcpp_VERSION_MINOR} LESS 8)
+if(${rclcpp_VERSION_MAJOR} LESS 1 AND ${rclcpp_VERSION_MINOR} LESS 8)
   message(STATUS "ROS Dashing detected")
   add_definitions(-DUSE_LEGACY_QOS_API)
 endif()

